authoringAssistant/src/main/java/explorableviz/transparenttext/Main.java

```diff
@@ -17,53 +17,74 @@ public static void main(String... args) {
         Map<String, String> arguments = parseArguments(args);
         logger.info("Arguments passed from command line");
         logger.info(arguments.toString().replace(",", "\n"));
-        final ArrayList<QueryContext> queryContexts;
-        final LearningQueryContext learningQueryContext;
+        final ArrayList<Query> queries;
+        final LearningQuery learningQuery;
         final Optional<Integer> numQueryToExecute = arguments.containsKey("numQueryToExecute") ? Optional.of(Integer.parseInt(arguments.get("numQueryToExecute"))) : Optional.empty();
         try {
-            Settings.getInstance().loadSettings(arguments.get("settings"));
-            learningQueryContext = LearningQueryContext.importLearningCaseFromJSON(arguments.get("inContextLearningPath"), Integer.parseInt(arguments.get("numLearningCaseToGenerate")));
-            queryContexts = TestQueryContext.loadCases(arguments.get("testPath"), Integer.parseInt(arguments.get("numTestToGenerate")));
-            final int queryLimit = numQueryToExecute.orElseGet(queryContexts::size);
-            final ArrayList<String> results = execute(learningQueryContext, arguments.get("agent"), queryLimit, queryContexts);
-            if (computeAccuracy(results, queryContexts, queryLimit, Integer.parseInt(arguments.get("threshold")))) {
+            Settings.init(arguments.get("settings"));
```

Comment:
Why are these things passed as command-line arguments:

`inContextLearningPath` (why not fix this as `system-prompt.json`?)
`settings` (why not fix as `settings.json`?)
`testPath` (why not fix as `testCases`?)
 
Having a setting where you specify where the settings are set is probably one level of indirection too many ;)

Reply:
Are arguments, these field are not read by the settings.json but from the command line.

Reply:
What I meant was: why have them as arguments at all? Why not just assume they are constants, as suggested above. We don’t need a parameter that tells us where the settings are stored or where the system prompt is stored – we just need to open the file called `settings.json` (that’s why it called `settings.json`), and similarly for the system prompt. 

`testPath` could be dealt with in the same way, or if really necessary put into `settings.json`.